lib/internal/worker.js

```diff
@@ -110,7 +111,13 @@ class Worker extends EventEmitter {
         );
       }
       url = null;
+      doEval = 'classic';
+    } else if (filename?.protocol === 'data:') {
```

Comment:
```suggestion
    } else if (isURLInstance(filename) && filename.protocol === 'data:') {
```

This should have a `isURLInstance` check to avoid unexpected types having their behavior being depended upon:

```mjs
new WorkerThread({
  protocol: 'data:'
}, ...);
```

And the test for such.